compiler/src/dmd/lexer.d

```diff
@@ -43,6 +37,18 @@ version (DMDLIB)
     version = LocOffset;
 }
 
+/***********************************************************
+ * Values to use for various magic identifiers
+ */
+struct CompileEnv
```

Comment:
Can we have this in `global` so we can later re-use it in other parts of the compiler ?

Reply:
I didn't put it in global because the idea is to isolate the lexer from global. I.e. be able to use a standalone lexer/parser without dragging in the rest of the compiler.

Reply:
I moved the initialization of CompileEnv into global because lexer should not be querying the operating system for date and time or anything else.

Reply:
But now `global` (and hence all of DMD) depends on `Lexer`, which is not ideal either. But I guess that can be broken down in another PR.

Reply:
Yes, I know. But Lexer is the only user of CompileEnv, and does it really work to put CompileEnv in its own file? Lexer needs to be able to refer to CompileEnv, but should not be responsible for initializing it. The reason is so that unittests can mock the inputs as required. All input to Lexer should be easily mockable for max testability.